allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -335,5 +269,54 @@ def deserialize_instance(self, string: str) -> Instance:
         The default implementation is to use `jsonpickle`.  If you would like some other format
         for your pre-processed data, override this method.
         """
-
         return jsonpickle.loads(string)  # type: ignore
+
+    def ensure_instance(self, raw_instance) -> Instance:
+        if isinstance(raw_instance, Instance):
+            return raw_instance
+        return self.text_to_instance(**raw_instance)
+
+    def _auto_islice(
```

Comment:
The name of this function could be clearer that it is uniquely applicable to distributed/multi-process data loading - maybe `multi_worker_islice`?

Also passing the transform is nice, but it could have a full docstring with an example, because it wasn't immediately clear what kind of transform I might pass here.